packages/core/src/transport/httpRetry.ts

```diff
@@ -0,0 +1,133 @@
+import type { TimeoutId } from '../tools/utils'
+import { ONE_SECOND } from '../tools/utils'
+import type { Observable } from '../tools/observable'
+import { monitor } from '../tools/monitor'
+
+const MIN_RETRY_INTERVAL = ONE_SECOND
+const MAX_RETRY_INTERVAL = 8 * ONE_SECOND
+
+export function httpRetry(
+  send: (data: string | FormData, bytesCount: number) => Observable<Response>,
+  configuration: {
+    bufferBytesLimit: number
+    ongoingBytesLimit: number
+    ongoingRequestsLimit: number
+  }
+) {
+  const buffer: Array<{ data: string | FormData; bytesCount: number }> = []
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Extracting a type for `{ data: string | FormData; bytesCount: number }` and also use it in send functions would be great